util/src/main/java/io/kubernetes/client/util/authenticators/ConfigPersister.java

```diff
@@ -0,0 +1,26 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package io.kubernetes.client.util.authenticators;
+
+import java.io.IOException;
+import java.util.Map;
+
+public interface ConfigPersister {
+    /**
+     * Save a new version of the configuration
+     * @param name The name of the authenticator configuration (e.g. 'azure')
+     * @param config The new configuration to persist
+     * @throws IOException when an error occurs while persisting
+     */
+    void persistConfig(String name, Map<String, Object> config) throws IOException;
```

Comment:
Why is the authenticator name required? It seems strange that you pass the `ConfigPersister` to `getToken` in the `Authenticator` which will always have to supply its own name to `persistConfig`.